cupy/lib/_routines_poly.py

```diff
@@ -13,8 +16,11 @@ def _get_coeffs(x):
 
     def wrapper(*args):
         coeffs = [_get_coeffs(x) for x in args]
-        out = func(*coeffs)
-
+        if func.__name__ == 'polymul':
+            out = func(*coeffs)
+        else:
+            with cupyx.allow_synchronize(False):
+                out = func(*coeffs)
```

Comment:
I decided not to continue with this to keep implementation and testing separated